astropy/nddata/bitmask.py

```diff
@@ -0,0 +1,411 @@
+good_mask_value=False, dtype=numpy.bool_)
+    Converts an array of bit fields to a boolean (or integer) mask array
+    according to a bitmask constructed from the supplied bit flags (see
+    ``ignore_flags`` parameter).
+
+    This function is particularly useful to convert data quality arrays to
+    boolean masks with selective filtering of DQ flags.
+
+    Parameters
+    ----------
+    bitfield : numpy.ndarray
+        An array of bit flags. By default, values different from zero are
+        interpreted as "bad" values and values equal to zero are considered
+        as "good" values. However, see ``ignore_flags`` parameter on how to
+        selectively ignore some bits in the ``bitfield`` array data.
+
+    ignore_flags : int, str, list, None (Default = 0)
+        An integer bitmask, a Python list of bit flags, a comma- or
+        '+'-separated string list of integer bit flags that indicate what
+        bits in the input ``bitfield`` should be *ignored* (i.e., zeroed), or
+        `None`.
+
+        | Setting ``ignore_flags`` to `None` effectively will make
+          `bitfield_to_boolean_mask` interpret all ``bitfield`` elements
+          as "good" regardless of their value.
+
+        | When ``ignore_flags`` argument is an integer bitmask, it will be
+          combined using bitwise-NOT and bitwise-AND with each element of the
+          input ``bitfield`` array (``~ignore_flags & bitfield``). If the
+          resultant bitfield element is non-zero, that element will be
+          interpreted as a "bad" in the output boolean mask and it will be
+          interpreted as "good" otherwise. ``flip_bits`` parameter may be used
+          to flip the bits (``bitwise-NOT``) of the bitmask thus effectively
+          changing the meaning of the ``ignore_flags`` parameter from "ignore"
+          to "use only" these flags.
+
+        .. note::
+
+            Setting ``ignore_flags`` to 0 effectively will assume that all
+            non-zero elements in the input ``bitfield`` array are to be
+            interpreted as "bad".
+
+        | When ``ignore_flags`` argument is a Python list of integer bit
+          flags, these flags are added together to create an integer bitmask.
+          Each item in the list must be a flag, i.e., an integer that is an
+          integer power of 2. In order to flip the bits of the resultant
+          bitmask, use ``flip_bits`` parameter.
+
+        | Alternatively, ``ignore_flags`` may be a string of comma- or
+          '+'-separated list of integer bit flags that should be added together
+          to create an integer bitmask. For example, both ``'4,8'`` and
+          ``'4+8'`` are equivalent and indicate that bit flags 4 and 8 in
+          the input ``bitfield`` array should be ignored when generating
+          boolean mask.
+
+        .. note::
+
+            ``'None'``, ``'INDEF'``, and empty (or all white space) strings
```

Comment:
@mwcraig I almost missed this issue... Support for string ``'None'`` and ``'INDEF'`` values is here because of backward compatibility that we need for HST software (and ``'INDEF'`` comes from IRAF). IMO we should not support this (I can try to take care of this this compatibility in our software outside of the `bitmask` module) in `astropy`. What is the policy here and what is your opinion? While this support is already implemented and so no extra effort is necessary, I am afraid to set a possibly bad precedent.

Reply:
Let's remove the support for 'None' and 'INDEF'; I don't know that there is a policy, but using `None` or an empty string seems clearer.